### ISSUE TITLE
Error with addon when trying to review

### ISSUE THREAD

**Issue by soccerwithag** [XavierScherschligt/RuneScape-Gamification#4]
# Getting an Error. Running on silicon monterey 12.6.1 macOS. Anki 2.1.55. Qt6

Error
An error occurred. Please start Anki while holding down the shift key, which will temporarily disable the add-ons you have installed.
If the issue only occurs when add-ons are enabled, please use the Tools > Add-ons menu item to disable some add-ons and restart Anki, repeating until you discover the add-on that is causing the problem.
When you've discovered the add-on that is causing the problem, please report the issue to the add-on author.
Debug info:
Anki 2.1.55 (01caec2a) Python 3.9.15 Qt 6.3.2 PyQt 6.3.1
Platform: macOS-12.6.1-arm64-arm-64bit
Flags: frz=True ao=True sv=3
Add-ons, last update check: 2022-12-29 20:38:37
Add-ons possibly involved: ⁨RuneScape-Gamification⁩

Caught exception:
Traceback (most recent call last):
  File "aqt.webview", line 562, in handler
  File "aqt.reviewer", line 655, in _onTypedAnswer
  File "aqt.reviewer", line 417, in _showAnswer
  File "_aqt.hooks", line 3761, in __call__
  File "anki.hooks", line 34, in runHook
  File "/Users/home/Library/Application Support/Anki2/addons21/931346754/__init__.py", line 130, in on_show_answer
    increase_skill_progress("speed", speed_xp)
  File "/Users/home/Library/Application Support/Anki2/addons21/931346754/__init__.py", line 93, in increase_skill_progress
    animate_xp_gain(skill)
  File "/Users/home/Library/Application Support/Anki2/addons21/931346754/__init__.py", line 71, in animate_xp_gain
    skill_label.setStyleSheet("color: orange;")
AttributeError: 'NoneType' object has no attribute 'setStyleSheet'

**Comment by XavierScherschligt** [comment]
Thanks for all the information! While the error references line 71 with the setStyleSheet method, I believe it's actually happening a bit early in the following function:

def animate_xp_gain(skill): 
     # get the dock widget 
     dock = mw.findChild(QtWidgets.QDockWidget) 
     # get the label that displays the skill 
     skill_label = dock.findChild(QtWidgets.QLabel, f"{skill}_label") 
     # change the text color of the label to orange 
     skill_label.setStyleSheet("color: orange;")

The code here attempts to find the docker that houses the skill labels and then find a specific skill on that docker and I think the real problem is happening during the findChild call to get the dock. Looking at my Anki brew, I have: Anki 2.1.55 (01caec2a) Python 3.9.15 Qt 6.4.0 PyQt 6.4.0. 

I see a small discrepancy in our Qt types (you have Qt 6.3.2) and I think that the call I was using hadn't been introduced in your version yet. Would you mind trying to download Anki again and see if that updates your Qt version to 6.4.0? I'm hoping that there isn't a slight version different on Mac compared to Windows and Ubuntu. I'll keep investigating the issue in the meantime.

TLDR: Try downloading Anki again and see if that updates your Qt version to 6.4.0, I'll keep looking into the issue.

Best, 
XS

**Comment by soccerwithag** [comment]
Looks like there is a discrepancy between mac and windows versions, I redownloaded and it shows the same version of qt. 

Edit: Sorry, this may be an issue due to pyenv and brew on my end, not sure. Will update.

**Comment by XavierScherschligt** [comment]
Well that's annoying. Thanks for trying though. I'll look into a workaround that will be compatible for both versions and post here if/when i find something.

Best,

XS

**Comment by soccerwithag** [comment]
Looks like it is not due to pyenv and brew, I think mac os build is still using PyQt 6.3.1 (see https://github.com/ankitects/anki/tree/main/python)

**Comment by dave-cao** [comment]
Hey, I'm getting the exact same error but on Linux. My Anki is using PyQt 6.4.0 so that might not be the problem.

```
Debug info:
Anki 2.1.57 (6b7d372c) Python 3.9.15 Qt 6.4.2 PyQt 6.4.0
Platform: Linux-5.17.5-76051705-generic-x86_64-with-glibc2.31
Flags: frz=True ao=True sv=3
Add-ons, last update check: 2023-02-06 00:24:08
Add-ons possibly involved: ⁨RuneScape-Gamification⁩

Caught exception:
Traceback (most recent call last):
  File "aqt.webview", line 562, in handler
  File "aqt.reviewer", line 644, in _onTypedAnswer
  File "aqt.reviewer", line 416, in _showAnswer
  File "_aqt.hooks", line 3790, in __call__
  File "anki.hooks", line 34, in runHook
  File "/home/cow/.local/share/Anki2/addons21/931346754/__init__.py", line 130, in on_show_answer
    increase_skill_progress("speed", speed_xp)
  File "/home/cow/.local/share/Anki2/addons21/931346754/__init__.py", line 93, in increase_skill_progress
    animate_xp_gain(skill)
  File "/home/cow/.local/share/Anki2/addons21/931346754/__init__.py", line 71, in animate_xp_gain
    skill_label.setStyleSheet("color: orange;")
AttributeError: 'NoneType' object has no attribute 'setStyleSheet'
```

Edit: I tested it on my Windows 10 machine and it worked perfectly. This might be a problem specifically with Linux/Mac

Edit2: Tried to do a little debugging. 
In the following function:
```python
def animate_xp_gain(skill):
    # get the dock widget
    dock = mw.findChild(QtWidgets.QDockWidget)
    # get the label that displays the skill
    skill_label = dock.findChild(QtWidgets.QLabel, f"{skill}_label")
    # change the text color of the label to orange
    skill_label.setStyleSheet("color: orange;")
    # create a thread to update the position of the skill label
    thread = threading.Thread(target=skill_label_color_change, args=(skill_label, 3))
    thread.start()  # start the thread
```
`dock` returns a proper object. It seems to be `skill_label = dock.findChild(QtWidgets.QLabel, f"{skill}_label")` that is the problem and returns None.

**Comment by dave-cao** [comment]
I figured out the problem on my machine. In my above comment, `skill_label` was returning `None`. I changed all instances of `dock.findChild` to `mw.findChild` and that fixed it for me. I made a pull request, #5 

@soccerwithag I tested it on Windows 10 and Linux but not Mac. Try it to see if it works for you.

**Comment by soccerwithag** [comment]
Disabling the progress bar addon cleared the error for me, so it may have something to do with how the dock was hooked into

**Comment by dave-cao** [comment]
Huh, I guess it worked for both of us then. That's good.
